vnext/Microsoft.ReactNative/Modules/AccessibilityInfoModule.cpp

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "AccessibilityInfoModule.h"
+#include <UI.Xaml.Automation.Peers.h>
+#include <UI.Xaml.Controls.h>
+#include <XamlUtils.h>
+#include <winrt/Windows.ApplicationModel.DataTransfer.h>
+#include <winrt/Windows.UI.ViewManagement.h>
+#include "Unicode.h"
+#include "Utils/Helpers.h"
+
+namespace Microsoft::ReactNative {
+
+void AccessibilityInfo::Initialize(winrt::Microsoft::ReactNative::ReactContext const &reactContext) noexcept {
+  m_context = reactContext;
+}
+
+void AccessibilityInfo::isReduceMotionEnabled(std::function<void(React::JSValue const &)> const &onSuccess) noexcept {
+  auto jsDispatcher = m_context.JSDispatcher();
+  m_context.UIDispatcher().Post([weakThis = weak_from_this(), jsDispatcher, onSuccess] {
+    if (auto strongThis = weakThis.lock()) {
+      winrt::Windows::UI::ViewManagement::UISettings uiSettings;
+      auto animationsEnabled = uiSettings.AnimationsEnabled();
+      jsDispatcher.Post([animationsEnabled, onSuccess] { onSuccess(!animationsEnabled); });
+    }
+  });
+}
+
+void AccessibilityInfo::isTouchExplorationEnabled(
+    std::function<void(React::JSValue const &)> const &onSuccess) noexcept {
+  onSuccess(false);
+}
+
+void AccessibilityInfo::setAccessibilityFocus(double /*reactTag*/) noexcept {
```

Comment:
>AccessibilityInfo::setAccessibilityFocus [](start = 5, length = 40)

We can use AutomationPeer::SetFocusCore() to implement this.  See:
[https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.setfocuscore?view=winrt-19041](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.automation.peers.automationpeer.setfocuscore?view=winrt-19041)
Also see XAML code that implements it here:
%SDXROOT%\onecoreuap\windows\dxaml\xcp\dxaml\lib\winrtgeneratedclasses\AutomationPeer.g.cpp

Reply:
That moves actual focus to that element.  This API is just trying to move the narrator focus.  -- So, when running narrator, it would move the blue rectangle without moving keyboard focus.

(Otherwise, this API is pointless, since there is a similar API that sets normal focus)

Reply:
Oops I didn't realize it's just supposed to move narrator focus.  I'm asking around, meanwhile I see some hits in Windows that suggest raising the AutomationFocusChanged event might work (eg %SDXROOT%\xbox\shellapps\Shared\Xbox.Foundation.UI\Narrator\NarratorReader.cpp line 125)

            // Raise the focus changed automation event, to ensure that this element receives narrator focus so that it reads the element and textToRead together.
            auto automationPeer = FrameworkElementAutomationPeer::FromElement(focusedElement);
            if (automationPeer != nullptr)
            {
                automationPeer.RaiseAutomationEvent(AutomationEvents::AutomationFocusChanged);
            }


---
In reply to: [481482991](https://github.com/microsoft/react-native-windows/pull/5849#discussion_r481482991) [](ancestors = 481482991)